### PR TITLE
:sparkles: feat(search): enable chat log search results

### PR DESCRIPTION
**Pull request by TreyDong**
#### 💻 Change Type

- [x] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 👷 build
- [ ] ⚡️ perf
- [ ] ✅ test
- [ ] 📝 docs
- [ ] 🔨 chore

🔀 Description of Change
Implemented global chat-log search results alongside assistant hits, wiring pagination through the message model, services, and TRPC router. Added a client-side results panel with in-context navigation via the Virtuoso list, plus updated English/Chinese locales and regression tests for the new query API.

📝 Additional Information

Message searches now use case-insensitive matching and return paging metadata (MessageKeywordSearchResult).
Legacy Dexie client search mimics the same interface for local builds.
The search sidebar only renders the assistant list when relevant results exist to avoid visual clutter.
A pre-existing bun dependency is still required to run targeted Vitest suites locally.

## Summary by Sourcery

Enable global chat log search with paginated results and integrate it into the chat UI

New Features:
- Add paginated chat log search support in database model, services, and TRPC router
- Introduce SearchMessages component with infinite scroll, snippet highlighting, and navigation hook
- Integrate chat log search results panel alongside session list in the chat sidebar

Enhancements:
- Implement case-insensitive matching for message searches
- Hide assistant list when no relevant session search results

Documentation:
- Update English and Chinese locale strings for search placeholders and labels

Tests:
- Add regression tests for keyword search API covering pagination and empty-input scenarios

### CODE REVIEW

File: packages/database/src/models/message.ts [hunk 1]
```diff
@@ -314,17 +312,57 @@ export class MessageModel {
...
+    { current = 0, pageSize = 20 }: { current?: number; pageSize?: number } = {},
+  ): Promise<MessageKeywordSearchResult> => {
+    const sanitizedKeyword = keyword.trim();
+    const safePageSize = pageSize > 0 ? pageSize : 20;
+    const safeCurrent = current >= 0 ? current : 0;
+
+    if (!sanitizedKeyword) {
+      return {
+        data: [],
+        pagination: {
```
Comment (arvinxx) [inline packages/database/src/models/message.ts:329]: pagination is a router result, model layer should only return data.

File: src/services/message/_deprecated.ts
Comment (arvinxx) [inline src/services/message/_deprecated.ts:1]: please remove it

**Review by arvinxx — CHANGES_REQUESTED** [review]
Thx! please rebase the latest next and then fix the review

**Comment by arvinxx** [comment]
@TreyDong Please rebase this, and also we now have CMD + J to invoke the quick panel, you can consider integrating the message search feature here~

image

---
> This comment was translated by Claude.

<details>
<summary>Original Content</summary>

@TreyDong 请 rebase 一下，另外我们现在已经有了 CMD + J 可以唤起快捷面板,可以考虑把搜索消息的能力做到这里来~

image

</details>

**Comment by TreyDong** [comment]
@arvinxx Okay, I'll adjust it

---
> This comment was translated by Claude.

<details>
<summary>Original Content</summary>

@arvinxx 好的 我来调整一下

</details>
